ipq806x: Fix default MAC addresses on MR52 by moving to `nvmem-layout`

rkboni opened the pull request:
ipq806x: Fix default MAC addresses on MR52 by moving to `nvmem-layout`

Partial, single-target update extracted from [openwrt#d264d3a6](https://github.com/openwrt/openwrt/commit/d264d3a6)

The previous `mac-address-increment` is deprecated, and in particular on
this target means that the kernel is unable to read the MAC address,
causing the system to boot with a new random MAC address each time.

Fixes: https://github.com/openwrt/openwrt/issues/15238

Signed-off-by: Rosen Penev <rosenp@gmail.com>
[rafal.boni@gmail.com: single-target-specific backport from larger change]
Signed-off-by: Rafal Boni <rafal.boni@gmail.com>

rkboni commented:
CC @neheb @Ansuel

Ansuel commented:
Ehhhh these kind of backport are a bit problematic and might introduce more regression... honestly it's better to fix the single target

namiltd commented:
@rkboni Look at: https://github.com/openwrt/openwrt/pull/16775
Can you test the snapshot version?

rkboni committed:
ipq806x: Fix default MAC addresses on MR52 by moving to `nvmem-layout`

Partial, single-target update extracted from https://github.com/openwrt/openwrt/commit/d264d3a6

The previous `mac-address-increment` is deprecated, and in particular on
this target means that the kernel is unable to read the MAC address,
causing the system to boot with a new random MAC address each time.

Fixes: https://github.com/openwrt/openwrt/issues/15238

Signed-off-by: Rosen Penev <rosenp@gmail.com>
[rafal.boni@gmail.com: single-target-specific backport from larger change]
Signed-off-by: Rafal Boni <rafal.boni@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16774
Signed-off-by: Robert Marko <robimarko@gmail.com>

rkboni commented:
@Ansuel seems like a reasonable objection -- trimmed this back to the one platform I know is broken and which I've tested on.

rkboni commented:
FWIW, I bought a couple of MR42's on eBay, so will also check this out on the MR42 once I get it (maybe a week or so) and either add the appropriate backport here or open up a new PR for it if this gets merged.

rkboni commented:
CC @robimarko since you merged the other MR52 PR, this is a more user-impacting one please 👀 ! 🙇

robimarko commented:
Thanks! Rebased on top of openwrt-23.05 and merged!
